### PR TITLE
Adding Utility isPositiveInteger function + test

### PR DESCRIPTION
**Pull request by devanp92**

### CODE REVIEW

**Comment by BigFav** [comment]
Nice idea, I was going to do this today. Several comments:
* The BigNumber portion should go in `util/BigNumber.js`.
* Since these are util functions, I believe it is reasonable to remove the type-checking as in-use this will cause a double check; each function should be 1 line.
* While the name `isPositiveInteger` is inherited, it is technically incorrect as it is 0 inclusive.

File: lib/util/number.js [hunk 1]
```diff
@@ -257,3 +257,21 @@ exports.nearlyEqual = function(x, y, epsilon) {
...
+* @param {Number} n   Value to determine if it is a positive integer
+* @return {Boolean}   Whether the number is positive
+*/
+exports.isPositiveInteger = function(n) {
+  var BigNumber = require('../type/BigNumber');
+  if (this.isNumber(n) && this.isInteger(n) && n >= 0) {
+    return true;
+  }
+
+  if (n instanceof BigNumber && n.isInteger() && n.gte(0)) {
```
Comment (BigFav) [inline lib/util/number.js]: I personally prefer `n.isInteger() && !n.isNegative()`, which saves creating a BigNumber object, but begs the question of how to handle -0.

Comment (josdejong) [inline lib/util/number.js]: I'm not sure whether it matters for handling `-0` (it's easy to add a test case for this though). 

But looking at the code of `isNegative` it is faster than using `gte`: it only checks the sign, whilst `gte` does a full comparison of all digits. So it may be nicer to use `isNegative` anyway.

**Comment by josdejong** [comment]
Thanks for the PR (including tests) @devanp92 . Can you please move the function from `/lib/util/number.js` to `/lib/util/bignumber.js` ?

File: lib/util/number.js (hunk 1 above)
Comment (devanp92) [inline lib/util/number.js]: I agree that 0 is not a positive integer. I have fixed this. I am moving this into bignumber.js.

**Comment by devanp92** [comment]
Ok. I have moved the functions to bignumber. I have fixed it to make sure that it excludes 0 and -0.

**Comment by josdejong** [comment]
Thanks Devan. Are you sure you don't need to include zero itself? Do the combinators functions you've created support zero?

**Comment by BigFav** [comment]
@josdejong Why should the check for type Number in `bignumber.js` instead of `number.js`? That makes no sense to me, so much so I would say it is incorrect (as opposed to a design decision).

**Comment by josdejong** [comment]
Ha ha I had never expected that such a little util function could spur so much discussion :)

My thought was like:
- This function `isPositiveInteger` very small and just by only two or three functions. You could inline a copy of it in each of these functions but it would be more neat to have the functions share the same `isPositiveInteger` via a util file (DRY).
- This function `isPositiveInteger` doesn't belong in `number.js` nor in `bignumber.js`, as it supports both `number` and `BigNumber`, and maybe in the future `Fraction` too, and others. But since it's such a small utility function let's not be too puristic right now, the overhead of putting it in it's own file may be not worth it right now. 
- From these two data types, `number` is the low level one and `BigNumber` is the high level one. Util methods of `number.js` should not have any notion of higher level types like `BigNumber`, it should only know about `number` and other primitives. `bignumber.js` kan use both numbers and BigNumbers, as `number` is lower level than `BigNumber` (many util functions in `bignumber.js` do use `number` arguments, which is ok). So from these two, `isPositiveInteger` fits best in `bignumber.js` right now.
- Also, doing a `require('../type/BigNumber')` inside the low level `number.js` would defeat the possibility to modularize the code (as we aim for with v2). If you would like to create a minimal bundle of math.js just supporting the lowest level data types, numbers, you still would get the whole BigNumber library included as it's referenced by `number.js`. Now you could circumvent this by using ducktyping instead of `instanceof` to check for a `BigNumber`, but qua architecture you keep having the dependency on `BigNumber` from `number.js`, which shouldn't be the case.

**Comment by BigFav** [comment]
My comment was to separate them into 2 functions (see my 2nd bullet point); for me it's the double type-checking that makes me dislike it, but I guess there is a balance of that and handling 2 imports in `lib/function/*/*`.

>Ha ha I had never expected that such a little util function could spur so much discussion :)

This just made me think: Is this overkill? I just looked up how many places this is currently used, and so far it's only in 2 files, nothing like `isInteger` which is in 24 files.

>I'm not sure whether it matters for handling -0 (it's easy to add a test case for this though).

All it takes is a `n.isZero()` haha. Maybe we should link the BigNumber API somewhere. (I know this is a late response haha)

**Comment by josdejong** [comment]
> This just made me think: Is this overkill? 

Same feeling here ;) 

I'm fine with either putting this util function `isPositiveInteger` in `bignumber.js`, or having it inlined twice in the functions that use it.

**Comment by devanp92** [comment]
I agree with @BigFav. I think isPositiveInteger is a bit overkill, but if implemented should be split into number.js and bignumber.js. In number.js it should only do type checking for isNumber and not bignumber, whereas in bignumber.js we should only type check for bignumber.

**Comment by josdejong** [comment]
Ok so we all agree this is overkill :) anway, we can leave the function in bignumber.js for now?

@devanp92 as for my other question: Is it correct that you exclude zero itself? Do the combinatorics functions you've created support `x >= 0` or `x > 0` ?

**Comment by devanp92** [comment]
It is correct to exclude zero [see here](mathworld.wolfram.com/PositiveInteger.html). Some functions require positive integers, some include 0. I will look into those that need 0.

**Comment by josdejong** [comment]
Ok then. So maybe you need an additional [`isNonNegativeInteger`](https://github.com/josdejong/mathjs/blob/develop/lib/function/probability/factorial.js#L100-L107) as well :D.

I will merge your PR now, we can extend on this later when needed. Thanks!

**Comment by josdejong** [comment]
@devanp92 I've reworked the functions you recently wrote (`stirlingS2`, `bellNumbers`, `composition`, and `multinomial`) for math.js `v2` (which is now in the `develop` branch, close to a release). I also introduced some new util functions: `isInteger`, `isNegative`, `isNumeric`, `isPositive`, and `isZero`. So now we can replace the `isPositiveInteger` functions with these fully fledged functions, which also include support for Fractions for example :D. 

Can you maybe replace the usages of `isPositiveInteger` in your functions with these new helper functions and add unit tests for edge cases like `x=0`? (these tests are still missing!) Would be great.
